client/db/bolt/db.go

```diff
@@ -818,16 +819,30 @@ func newestBuckets(master *bbolt.Bucket, n int, timeKey []byte, filter func(*bbo
 	return idx.pairs
 }
 
-// makeTopLevelBuckets creates a top-level bucket for each of the provided keys,
-// if the bucket doesn't already exist.
-func (db *boltDB) makeTopLevelBuckets(buckets [][]byte) error {
+// init sets the database version and creates top-level bucket
+// for each of the provided keys, if the bucket doesn't already exist.
+func (db *boltDB) init(buckets [][]byte) error {
 	return db.Update(func(tx *bbolt.Tx) error {
 		for _, bucket := range buckets {
 			_, err := tx.CreateBucketIfNotExists(bucket)
 			if err != nil {
 				return err
 			}
 		}
+
+		// Set the db version if it does not already exist.
+		bucket := tx.Bucket(appBucket)
+		if bucket == nil {
+			return fmt.Errorf("app bucket not found")
+		}
+		version := bucket.Get(versionKey)
+		if version == nil {
+			err := bucket.Put(versionKey, encode.Uint32Bytes(uint32(DBVersion)))
```

Comment:
Correct me if I'm wrong.
This will always set the initial version as the current version.
I think in dcrwallet, the initial version is always set as 0, and upgrades take place for new databases.
I don't know why that is. Is this way better, or is there a reason dcrwallet always goes through all updates from the initial version?

Reply:
Currently it'd set the initial version as the current version however when upgrades are made `DBversion` will be updated. Since the highlighted line is only going to execute for a new db I don't think it is necessary to run through all the the upgrades, it'd be empty. Probably prematurely optimizing here, should run through all the upgrades.

Reply:
Here is the log when creating a new dcrwallet:
```
Creating the wallet...
2020-06-08 16:09:46.260 [INF] WLLT: Upgrading database from version 1 to 15
2020-06-08 16:09:48.360 [INF] WLLT: Opened wallet
The wallet has been created successfully.
```
Which looks like the database is updated on New as well. I haven't looked at the code recently though, so I'm not sure. But if it is I wonder what the reason is.

Reply:
in dcrwallet init https://github.com/decred/dcrwallet/blob/13390d5c216ad1dfe389e7c95d4a9a7bb57b0719/wallet/udb/initialize.go#L58